Aplib.Tests/Belief/MemoryBeliefTests.cs

```diff
@@ -54,22 +55,22 @@ public void GetMemoryAt_WhenObservationIsUpdated_ShouldReturnObservationAtSpecif
     /// <summary>
     /// Given a MemoryBelief instance with an observation,
     /// When asking for an index that is out of bounds,
-    /// Then the closest element that is in bounds is returned.
+    /// Then an exception should be thrown.
     /// </summary>
     [Fact]
-    public void GetMemoryAt_IndexOutOfBounds_ShouldReturnClosestElement()
+    public void GetMemoryAt_IndexOutOfBounds_ShouldThrowException()
     {
         // Arrange
         List<int> list = [1, 2, 3];
         MemoryBelief<List<int>, int> belief = new(list, reference => reference.Count, 3);
 
         // Act
-        list.Add(4);
-        belief.UpdateBelief();
+        void GetMemoryAtNegativeIndex() => belief.GetMemoryAt(-1);
+        void GetMemoryAtIndexGreaterThanCount() => belief.GetMemoryAt(3);
 
         // Assert
-        Assert.Equal(3, belief.GetMemoryAt(-1, true));
-        Assert.Equal(0, belief.GetMemoryAt(5, true));
+        Assert.Throws<ArgumentOutOfRangeException>(GetMemoryAtNegativeIndex);
+        Assert.Throws<ArgumentOutOfRangeException>(GetMemoryAtIndexGreaterThanCount);
```

Comment:
Please write two separate tests for this. Asserting multiple things in a test is a test smell, as it is difficult to quickly pinpoint what the point of failure is.

In general, I'd think multiple asserts per test are allowed if you need to assert multiple thing with the same setup (arrange and act). This method uses two act methods and thus do not comply.